org.osate.ge/src/org/osate/ge/internal/diagram/runtime/DiagramSerialization.java

```diff
@@ -50,13 +60,25 @@
 		}
 	}
 
-	public static AgeDiagram createAgeDiagram(final org.osate.ge.diagram.Diagram mmDiagram) {
+	public static AgeDiagram createAgeDiagram(final org.osate.ge.diagram.Diagram mmDiagram,
+			final ExtensionRegistryService extRegistry) {
+		Objects.requireNonNull(extRegistry, "extRegistry is null");
+
 		// Set the id which should be used for new diagram elements.
 		final AgeDiagram ageDiagram = new AgeDiagram(getMaxIdForChildren(mmDiagram) + 1);
 
 		// Read the diagram configuration
-		final DiagramConfigurationBuilder configBuilder = new DiagramConfigurationBuilder(
-				ageDiagram.getConfiguration());
+
+		// Set the diagram type
+		final DiagramType diagramType;
+		final String diagramTypeId = mmDiagram.getConfig() == null || mmDiagram.getConfig().getType() == null
+				? CustomDiagramType.ID
+						: mmDiagram.getConfig().getType();
+		diagramType = extRegistry.getDiagramTypeById(diagramTypeId)
+				.orElseGet(() -> new UnrecognizedDiagramType(diagramTypeId));
```

Comment:
Is there a reason diagramType is created above where it is initialized?

Reply:
No, that is likely just an artifact of some of the revisions while writing the code. Just pushed a fix.